tests/regression/09-regions/26-alloc_region_rc.c

```diff
@@ -49,12 +49,13 @@ int main () {
   pthread_mutex_lock(&A_mutex);
   insert(p, &A);
   pthread_mutex_unlock(&A_mutex);
-  
+
   pthread_mutex_lock(&A_mutex);
   p = A;
-  while (p->next)
+  while (p->next) {
```

Comment:
This changes the semantics of the test program. Is this on purpose?

Reply:
This is on purpose because the original program was actually race-free, although for very complicated reasons (for Goblint). This loop in the main thread would've always moved to the last element of the singly linked list to access its data. The other thread would've modified the data of `p` which it prepended to the same singly linked list, but `p` could never be the last element of that list. We decided keeping that original program in our test suite would be worthless because we don't intend for Goblint to be able to do that complicated reasoning.

The first modification of this program also revealed what seems to be a different bug, so there's a skipped 09/33 test.